pybop/parameters/priors.py

```diff
@@ -38,12 +105,56 @@ def __init__(self, lower, upper):
         self.upper = upper
 
     def pdf(self, x):
+        """
+        Calculates the probability density function of the uniform distribution at x.
+
+        Parameters
+        ----------
+        x : float
+            The point at which to evaluate the pdf.
+
+        Returns
+        -------
+        float
+            The probability density function value at x.
+        """
         return stats.uniform.pdf(x, loc=self.lower, scale=self.upper - self.lower)
 
     def logpdf(self, x):
+        """
+        Calculates the logarithm of the pdf of the uniform distribution at x.
+
+        Parameters
+        ----------
+        x : float
+            The point at which to evaluate the log pdf.
+
+        Returns
+        -------
+        float
+            The log of the probability density function value at x.
+        """
         return stats.uniform.logpdf(x, loc=self.lower, scale=self.upper - self.lower)
 
     def rvs(self, size):
+        """
+        Generates random variates from the uniform distribution.
+
+        Parameters
+        ----------
+        size : int
+            The number of random variates to generate.
+
+        Returns
+        -------
+        array_like
+            An array of random variates from the uniform distribution.
+
+        Raises
+        ------
+        ValueError
+            If the size parameter is not positive.
+        """
         if size < 0:
```

Comment:
This check allows size=0 without error, is this intended?
If so, change to "If the size parameter is negative." (i.e. not positive and not zero).
Otherwise use `size < 1`.